drivers/rpmsg/rpmsg_syslog.c

```diff
@@ -0,0 +1,157 @@
+/* SPDX-License-Identifier: GPL-2.0 */
+/*
+ * Copyright (C) 2017 Pinecone Inc.
+ */
+
+#include <linux/module.h>
+#include <linux/rpmsg.h>
+#include <linux/slab.h>
+
+#define RPMSG_SYSLOG_TRANSFER		0
+#define RPMSG_SYSLOG_TRANSFER_DONE	1
+#define RPMSG_SYSLOG_SUSPEND		2
+#define RPMSG_SYSLOG_RESUME		3
+
+struct rpmsg_syslog_header {
+	uint32_t			command;
+	int32_t				result;
+} __packed;
+
+struct rpmsg_syslog_transfer {
+	struct rpmsg_syslog_header	header;
+	uint32_t			count;
+	char				data[0];
+} __packed;
+
+#define rpmsg_syslog_suspend		rpmsg_syslog_header
+#define rpmsg_syslog_resume		rpmsg_syslog_header
+#define rpmsg_syslog_transfer_done	rpmsg_syslog_header
+
+struct rpmsg_syslog {
+	char				*tmpbuf;
+	unsigned int			nextpos;
+	unsigned int			alloced;
+};
+
+static int rpmsg_syslog_callback(struct rpmsg_device *rpdev,
```

Comment:
some comments needed here explaining what this does

Reply:
the code is to handle this case:
the first mesage is "[    0.000000] sensor: NuttX version 7.26-686de48"
the second messgae is "-dirty (arm@song/banks) Oct 12\n"
we have to merge these two message into one to avoid the log interlace with other kernel log.
If the remote side implement the log buffer as a circle buffer, it most likely has to split the buffer and send in two ipc message at the wrapper pointer.